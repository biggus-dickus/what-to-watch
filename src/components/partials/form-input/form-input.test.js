import * as React from 'react';
import renderer from 'react-test-renderer';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import FormInput from './form-input';

configure({adapter: new Adapter()});


describe(`FormInput test suite`, () => {
  it(`<FormInput /> should render correctly`, () => {
    const mockField = {
      htmlFormTag: `input`,
      label: `Remember me`,
      type: `checkbox`,
      name: `remember-the-fallen`,
      id: `some-checkbox`
    };

    const tree = renderer.create(<FormInput {...mockField} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`should render textarea correctly`, () => {
    const mockField = {
      htmlFormTag: `textarea`,
      name: `feedback`,
      id: `impressions`
    };

    const wrapper = shallow(<FormInput {...mockField} />);
    expect(wrapper.find(`textarea`)).toHaveLength(1);
  });

  it(`should render select field correctly`, () => {
    const mockField = {
      htmlFormTag: `select`,
      name: `payment-type`,
      id: `payment-type`
    };

    const options = [`visa`, `mastercard`, `amex`];

    const wrapper = shallow(
        <FormInput {...mockField}>
          {options.map((it) => <option key={it}>{it}</option>)}
        </FormInput>
    );

    const select = wrapper.find(`select`);

    expect(select).toHaveLength(1);
    expect(select.children()).toHaveLength(options.length);
  });
});

