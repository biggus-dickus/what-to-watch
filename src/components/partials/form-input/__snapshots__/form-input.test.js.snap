// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput test suite <FormInput /> should render correctly 1`] = `
<input
  id="some-checkbox"
  label="Remember me"
  name="remember-the-fallen"
  type="checkbox"
/>
`;
